[Bug] [Build] Build fail with clang

### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.


### Version

master

### What's Wrong?
```cpp
bitshuffle_page.h:242:15: error: 'static' function 'parse_bit_shuffle_header' declared in header file should be declared 'static inline' [-Werror,-Wunneeded-internal-declaration]
```
### What You Expected?

fix

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/be/src/olap/rowset/segment_v2/bitshuffle_page.h
+++ b/be/src/olap/rowset/segment_v2/bitshuffle_page.h
@@ -219,7 +219,7 @@ private:
         return _buffer.build();
     }
 
-    typedef typename TypeTraits<Type>::CppType CppType;
+    using CppType = typename TypeTraits<Type>::CppType;
 
     CppType cell(int idx) const {
         DCHECK_GE(idx, 0);
@@ -239,7 +239,7 @@ private:
     CppType _last_value;
 };
 
-static Status parse_bit_shuffle_header(const Slice& data, size_t& num_elements,
+inline Status parse_bit_shuffle_header(const Slice& data, size_t& num_elements,
                                        size_t& compressed_size, size_t& num_element_after_padding,
                                        int& size_of_element) {
     if (data.size < BITSHUFFLE_PAGE_HEADER_SIZE) {
@@ -425,7 +425,7 @@ private:
         memcpy(data, get_data(_cur_index), n * SIZE_OF_TYPE);
     }
 
-    typedef typename TypeTraits<Type>::CppType CppType;
+    using CppType = typename TypeTraits<Type>::CppType;
 
     enum { SIZE_OF_TYPE = TypeTraits<Type>::size };
 

